decompiler/frontend/binaryninja/handlers/globals.py

```diff
@@ -12,30 +23,161 @@ class GlobalHandler(Handler):
-        parent: Optional[MediumLevelILInstruction] = None, **kwargs
-    ) -> UnaryOperation:
-        """Lift global variables with basic types (pointer are possible)"""
+        parent: Optional[MediumLevelILInstruction] = None, caller_addr: int = 0, **kwargs
+    ) -> Union[Symbol, UnaryOperation, GlobalVariable, StringSymbol]:
+        """Lift global variables via datavariable type. Check bninja error case + recursive datavariable first"""
+        if not self._addr_in_section(view, variable.address):
+            return Constant(variable.address, vartype=Integer(view.address_size*8, False))
+
+        if caller_addr == variable.address: 
+            return self._lifter.lift(variable.symbol) if variable.symbol else \
+            Symbol("data_" + f"{variable.address:x}", variable.address, vartype=Integer.uint32_t())
+
+        return self._datavariable_types[type(variable.type)](variable, view, parent)
+
+
+    def _lift_constant_type(self, variable: DataVariable, view: BinaryView, parent: Optional[MediumLevelILInstruction] = None) -> StringSymbol:
+        """Lift a constant datavariable directly into code (bninja should only yield strings)"""
+        string = str(variable.value)[2:-1].rstrip('\\x00')
+        return StringSymbol(f'"{string}"', variable.address, vartype=Pointer(Integer.char(), view.address_size * 8))
+
+    
+    def _lift_pointer_type(self, variable: DataVariable, view: BinaryView, parent: Optional[MediumLevelILInstruction] = None):
+        """Lift a pointer as:
+            1. Function pointer: If bninja already knows it's a fkt. ptr.
+            2. Type pointer: As normal type pointer (there _should_ be a datavariable at the pointers dest.)
+            3. Void pointer: Try to extract a datavariable (recover type of void* directly), string (char*) or raw bytes (void*) at the given address
+        """
+        if isinstance(variable.type.target, FunctionType):
+            return ImportedFunctionSymbol(variable.name, variable.address, vartype=Pointer(Integer.char(), view.address_size * 8))
+        if isinstance(variable.type.target, VoidType):
+            init_value, type = self._get_unknown_value(variable.value, view, variable.address)
+            if not isinstance(type, PointerType): # Fix type to be a pointer (happens when a datavariable is at the dest.)
+                type = Type.pointer(view.arch, type)
+        else:
+            init_value, type = self._lifter.lift(view.get_data_var_at(variable.value), view=view, caller_addr=variable.address), variable.type
         return UnaryOperation(
             OperationType.address,
                 [
                     GlobalVariable(
-                    variable.name if variable.name else "data_" + f"{variable.address:x}",
-                    self._lifter.lift(variable.type),
+                    name=self._lifter.lift(variable.symbol).name if variable.symbol else "data_" + f"{variable.address:x}",
```

Comment:
`GLOBAL_VARIABLE_PREFIX = "data_"`

Reply:
Done in symbols, because both files generate "data_"